### PR TITLE
Add debug docs

### PR DESCRIPTION
**Pull request by Ernegien**
This should serve as a good starting point for some developer documentation. There's lots more to do, and I'm still debating how much depth to cover, but definitely open to any early feedback. This will likely remain in draft status for a bit as I continue to flush things out, particularly the graphics section, which I'll probably yield to others.

### CODE REVIEW

File: docs/docs/dev/debug/guest.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+DEBUG_DSP | hw/xbox/mcpx/dsp/dsp.c | Audio
+
+<!--
+## Tracing
+
+A more advanced version of logging can be accomplished via qemu's tracing infrastructure. Documentation for such is located underneath `docs/devel/tracing.rst`. Review the `trace-events` files within the source tree to see what's supported.
+
+#### Runtime Usage (preferred)
+
+TODO:
```
Comment (abaire) [inline docs/docs/dev/debug/guest.md:65]: 0. Launch xemu from a console window (cmd.exe, Terminal.app, bash, etc...; whichever is appropriate for your OS)
1. Open the monitor via `Debug > Monitor` in the menu
2. Enter `trace-event <trace_event> on` to turn on an event, `trace-event <trace_event> off` to turn one off, replacing `<trace_event>` with the name of the event you wish to toggle.
3. Examine the console where you launched xemu, trace messages will be written to `STDERR`

The [nv2a events](https://github.com/mborgerson/xemu/blob/master/hw/xbox/nv2a/trace-events) are particularly interesting when debugging graphics issues, specifically the `nv2a_pgraph_*` subset. These may be toggled all at once using a wildcard. For example, `trace-event nv2a_pgraph_* on` will turn them all on.

See also [QEMU user documentation for tracing](https://qemu-project.gitlab.io/qemu/devel/tracing.html)

Comment (abaire) [reply docs/docs/dev/debug/guest.md:65]: Note: As of right now it looks like the log backend is only enabled in `--debug` builds https://github.com/mborgerson/xemu/blob/master/build.sh#L168

Comment (Ernegien) [reply docs/docs/dev/debug/guest.md]: These should now be added to my most recent commit.

**Review by abaire — APPROVED** [review]

File: docs/docs/dev/debug/index.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+The xemu source tree can be quite intimidating at first due to its qemu roots, but the majority of Xbox-specific code is confined to the `xemu/hw/xbox/` folder where the rest can be disregarded.
```
Comment (mborgerson) [inline docs/docs/dev/debug/index.md:1]: * nit: QEMU is stylized in all uppercase (in contrast with xemu which is stylized in all lowercase). Let's change mentions of QEMU in this PR to use the uppercase convention.
* Let's drop the `due to its qemu roots`
* Let's change `, but the majority of Xbox-specific code` to `. Xbox-specific hardware emulation code`
* Let's drop the `rest can be disregarded` line. It depends on what someone is fixing, and many things do live outside of hw/xbox, like all of the UI, core networking stuff, main loop, etc.

File: docs/docs/dev/debug/index.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+The xemu source tree can be quite intimidating at first due to its qemu roots, but the majority of Xbox-specific code is confined to the `xemu/hw/xbox/` folder where the rest can be disregarded.
+
+The basic principles are to emulate an original Xbox system at the hardware level via qemu intercepting memory-mapped IO (MMIO) intended for devices and re-implementing functionality as close to the original as possible. In the case of the NV2A GPU, these MMIO get translated to equivalent OpenGL instructions.
```
Comment (mborgerson) [inline docs/docs/dev/debug/index.md:3]: The gist is correct but the terminology is a bit loose.

`qemu intercepting` really means CPU emulation (which is provided by the QEMU project with improvements made by xemu project for Xbox emulation) properly routes memory accesses made to emulated hardware.

MMIO here means device access by CPU client via memory address space, which does happen for hardware configuration. However pushbuffer commands are actually consumed and executed by (emulated) hardware asynchronously via DMA. Then when processing those commands we emulate the behavior of the 2D/3D graphics hardware using OpenGL API on your host machine. There are parallels between hardware commands and OpenGL API, but it is not just a simple 1:1 map, which was my interpretation of the wording.

Let's re-word it for clarity/simplify, or we can just drop this paragraph for now and add details later.

File: docs/docs/dev/debug/guest.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+Open the monitor via <kbd>Debug</kbd> &rarr; <kbd>Monitor</kbd> in the xemu menu.
+
+Enter `trace-event <trace_event> on` to turn on an event, `trace-event <trace_event> off` to turn one off, replacing `<trace_event>` with the name of the event you wish to toggle.
+
+Examine the console where you launched xemu; trace messages will be written to `STDERR`.
+
+The [nv2a events](https://github.com/mborgerson/xemu/blob/master/hw/xbox/nv2a/trace-events){target=_blank} are particularly interesting when debugging graphics issues, specifically the `nv2a_pgraph_*` subset. These may be toggled all at once using a wildcard. For example, `trace-event nv2a_pgraph_* on` will turn them all on.
+
+#### Compile-Time Usage (deprecated)
```
Comment (mborgerson) [inline docs/docs/dev/debug/guest.md:75]: It's not deprecated (afaik), just a different backend

**Review by mborgerson — COMMENTED** [review]
Overall looks good, just some nits

**Comment by mborgerson** [comment]
Thanks!

**Comment by mborgerson** [comment]
Fixed the merge conflicts. Next time please rebase instead of merging

**Pull request merged**